playbooks/generic-tempest-output-to-junit.yml

```diff
@@ -0,0 +1,83 @@
+---
+- hosts: soc-deployer
+  gather_facts: yes
+  tasks:
+    - name: Run container to attach tempest pvc
+      shell: |
+        set pipefail -o
+        cat <<EOF | kubectl -n openstack apply -f -
+        apiVersion: v1
+        kind: Pod
+        metadata:
+          name: tempest-data
+        spec:
```

Comment:
This could use an init container that mounts the volume and runs cat /data/.stestr/0 | subunit-1to2 | subunit2junitxml > /data/tempest-junit.xml (for example, evrardjp/subunit2junit).

Reply:
I dont want to make this more complicated, and that sounds more complicated. Why have init containers to do that when I can do it locally?

Also, this would depend on a extra image, which means more data to download. Separating what the container does to make it minimal means we can move this image to be a really small one (alpine?) so the mount+extraction is really fast. This is already a pain in the ass (mounting a pod to extract data) so I want to minimize the impact of this as much as possible.

Also, I dont want to use containers that much for data processing as they can go away at any moment, this is just a help container to get the data from the pvc and extract it, processing can be done offline.

Reply:
You have to deal with dependency management, and installing stuff in a venv.
venv vs container, it's not really a big difference.

Here the host stays clean, it only needs kubectl.

Reply:
I dont want to use a personal image for this, it should be using a generic image available to all. But I also dont want to manage installing stuff inside a container. So unless there is an image supported by an org (say the official subunit container) that can guarantee that the image is well supported and its not going to break and so on, this is a big no for me.

This also means that we are hiding our processing. Part of it done on ansible, part of it doing on k8s. That just makes it more difficult to maintain and find any issues that arise.

Im totally 100% against this idea, sorry.

>You have to deal with dependency management, and installing stuff in a venv.
venv vs container, it's not really a big difference.

There is actually. Ansible has a module for this that guarantees several things (backwards compatibility, idempotency, etc) meanwhile the container+init does not and we have to manage that ourselves. Notonly that but for a simple task (create venv, install 2 packages, delete venv) we are introducing 3 more layers (k8s, containers, remote)

> Here the host stays clean, it only needs kubectl. In the future, that can even go away, and only have podman and shell.

Host is also clean here, plus it already has python running (otherwise ansible...) so create a venv and deleting it its clean enough. Also this is only for CI.

Reply:
> I dont want to use a personal image for this, it should be using a generic image available to all. But I also dont want to manage installing stuff inside a container.

This is for CI. We can build our own in IBS/OBS that wouldn't be a big deal.

> This also means that we are hiding our processing. Part of it done on ansible, part of it doing on k8s.

You are currently doing an ansible that runs a shell script that generates the k8s manifest to then apply it using kubectl. I am not sure this is worse, as this effectively teaches us the good practices of k8s, whether they suck or not.

> Im totally 100% against this idea, sorry.
> 
> > You have to deal with dependency management, and installing stuff in a venv.
> > venv vs container, it's not really a big difference.
> 
> There is actually. Ansible has a module for this that guarantees several things (backwards compatibility, idempotency, etc) meanwhile the container+init does not and we have to manage that ourselves. Notonly that but for a simple task (create venv, install 2 packages, delete venv) we are introducing 3 more layers (k8s, containers, remote)

We'll agree to disagree. There is a non zero count of persons that stumbled on the venv thing in this team.

> Host is also clean here, plus it already has python running (otherwise ansible...) so create a venv and deleting it its clean enough. Also this is only for CI.

If this is only for CI, why would it be a problem to make it a container?

Reply:
We have to remember of the end goal here. I am fine with ansible running for CI. I think at the end of the day SOC10 should not be exposing ansible plays, as the org is terrible with it (too old ansible, nobody would care except us, the state of the architecture of our plays and roles is just awful). If we can replace the whole series of tasks and keep the same feature for CI using native k8s objects, why wouldn't we do that?

We should merge this PR in any case, as it's good enough, and doesn't hurt the project, quite the contrary.